### ISSUE TITLE
[VL] Large timestamp outside of range

### ISSUE THREAD

**Issue by ccat3z** [apache/incubator-gluten#7109]
### Backend

VL (Velox)

### Bug description

Timestamps that great then 32767 years without but do not overflow will be outside of range.

``` scala
Seq(
  (1723240800675l)
).toDF("time")

spark.sql("""
  select from_unixtime(event_timestamp, 'HH') from test
""")

// vanilla: 56577-04-30 00:11:15
// gluten: Timestamp is outside of supported range of [-32767-01-01, 32767-12-31]
```

It may be because `std::chrono::time_point` is used in time conversion, which has range limitations.

### Spark version

None

**Comment by zml1206** [comment]
Seems to be a known issue https://github.com/facebookincubator/velox/pull/10641

**Comment by ccat3z** [comment]
> Seems to be a known issue [facebookincubator/velox#10641](https://github.com/facebookincubator/velox/pull/10641)

It maybe not same issue, https://github.com/facebookincubator/velox/pull/10641 still use std::chrono. But the expected result is 56577, which std::chrono not supported.

**Comment by ccat3z** [comment]
I

> > Seems to be a known issue [facebookincubator/velox#10641](https://github.com/facebookincubator/velox/pull/10641)
> 
> It maybe not same issue, [facebookincubator/velox#10641](https://github.com/facebookincubator/velox/pull/10641) still use std::chrono. But the expected result is 56577, which std::chrono not supported.

I am trying to remove the dependency on std::chrono to get rid of the limitations.

**Comment by wForget** [comment]
@NEUpanning  Does #7375 resolve this issue?

**Comment by NEUpanning** [comment]
> @NEUpanning Does https://github.com/apache/incubator-gluten/pull/7375 resolve this issue?

@wForget No. See task 'Fix date range validation in SimpleDateFormat standard' part in https://github.com/facebookincubator/velox/issues/10354
